EC 8+6 MSR rule test workflow changes

pdhiran opened the pull request:
Summary of Changes in the PR


1. Improved pool clean-up for EC pools, now deleting the CRUSH Rule & the EC Profile created for the EC pools during clean-up.
2. RBD init has been moved down in pool creation, this is because overwrites need to be enabled on EC pools, before we run init in case of RBD.
3. Added new module for deletion of EC profile, and printing the CRUSH rule used in creation of EC pools.
4. Added new "--yes-i-mean-it" flag support for modification of EC pools, introduced in 8.0.
5. Updated few commands to be run on Client Nodes.
6. Printing Ceph Status along with health detail in the finally block.
7. Created New conf file, For running 4 Node EC pool tests in Openstack env in Squid & Reef. `conf/reef/rados/4-node-ec-cluster-1-client.yaml` & `conf/squid/rados/4-node-ec-cluster-1-client.yaml`
8.  Created New conf file, For running 3 AZ tests in Openstack env. `conf/squid/rados/3AZ-cluster.yaml`
9. Added new Test case in all the releases, testing the EC profile behaviour. Polarion ID : CEPH-83596295 . Test code in `pool_tests.py` module
10. Updated the deployment suite for 4 Node EC pool clusters, to facilitate node removals & additions in future via CI.
11. Deployment Suite file for 3 AZ clusters with mon locations. `suites/squid/rados/tier-3_rados_test-3-AZ-Cluster.yaml`
12. Modified method ` get_device_path() `, to accommodate the host running containers without name or command. This was a issue in 4 node clusters, where containers like grafana, prometheus & alertmanager run colocated with OSD hosts.
13. Automated 3 bugzilla & tracker scenarios in Online reads balancer module, to check if the read suggestions are valid, are added in osdmap & to confirm the read suggestions are enforced on the clusters.
14. Updated the log line to be identified during scrub/ deep scrub scenarios.

neha-gangadhar commented:
@pdhiran Please add description, fix tox issues and attach logs if any run for feature testing is fine as well.

harshkumarRH commented on ceph/rados/core_workflows.py:
```diff
@@ -1112,9 +1112,40 @@ def delete_pool(self, pool: str) -> bool:
             log.error(f"Pool:{pool} does not exist on cluster, cannot delete")
             return True
 
+        pool_details = self.run_ceph_command(
+            cmd="ceph osd pool ls detail", client_exec=True
+        )
         cmd = f"ceph osd pool delete {pool} {pool} --yes-i-really-really-mean-it"
         self.client.exec_command(cmd=cmd, sudo=True)
 
+        time.sleep(2)
+        # Improving cleanup for EC pools
+        for pool in pool_details:
+            if pool["pool_name"] == pool:
```
We already have a method called `def get_pool_details(self, pool) -> dict:` in core workflows which would return the pool ls detail for the input pool name, we can use this an eliminate the additional traversal using for loop

pdhiran replied:
done

harshkumarRH commented on ceph/rados/pool_workflows.py:
```diff
@@ -740,7 +740,7 @@ def modify_autoscale_threshold(self, threshold: float):
         """
         cmd = f"ceph osd pool set threshold {threshold}"
-        self.rados_obj.run_ceph_command(cmd=cmd)
+        self.rados_obj.node.shell([cmd])
```
self.node is also initiated in this class, can we use it directly?

pdhiran replied:
Changed the call.

harshkumarRH commented on conf/squid/rados/4-node-ec-cluster-1-client.yaml:
```diff
@@ -0,0 +1,53 @@
+            - mgr
+            - rgw
+          no-of-volumes: 6
+          disk-size: 50
```
Is it necessary to have such large disks?

pdhiran replied:
Since we only have 4 nodes, I felt we should have enough storage space.

Updated the capacity to 25G as other configs from RADOS. will update this if needed in future.

pdhiran commented on ceph/rados/core_workflows.py:
```diff
@@ -1144,7 +1171,26 @@ def get_ec_profiles(self) -> list:
+            Profile deleted -> True
+            profile not deleted -> False
+        """
+        profile_del = f"ceph osd erasure-code-profile delete {profile}"
```
update the command to use rm

pdhiran commented on ceph/rados/core_workflows.py:
```diff
@@ -2119,7 +2169,7 @@ def get_pgid(
             cmd = f"{cmd} {states}"
 
-        pgid_dict = self.run_ceph_command(cmd=cmd)
+        pgid_dict = self.run_ceph_command(cmd=cmd, client_exec=True)
```
add the print for crush rule dump for the rule created for EC pools.

pdhiran commented on suites/reef/rados/tier-3_rados_test-4-node-ecpools.yaml:
```diff
@@ -383,8 +417,8 @@ tests:
-              conf: sample-pool-4
-        pool_configs_path: "conf/reef/rados/test-confs/pool-configurations.yaml"
+              conf: sample-pool-5
+        pool_configs_path: "conf/squid/rados/test-confs/pool-configurations.yaml"
```
update the path for the config file

pdhiran commented:
pass log:
3az cluster deployment : http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-EPH160

pdhiran commented on suites/squid/rados/tier-2_rados_test-pool-functionalities.yaml:
```diff
@@ -419,48 +419,3 @@ tests:
           - rpool_2
           - rpool_3
           - ecpool_test_2
-
```
Moved the read balancer tests with "read" mode to Clay EC profile tests suite.

pdhiran commented:
> @pdhiran Please add description, fix tox issues and attach logs if any run for feature testing is fine as well.

Fixed the issues and added summary of all the changes made in the PR.

SrinivasaBharath commented:
Please provide the bugzilla and tracker details for the following task.

Automated 3 bugzilla & tracker scenarios in Online reads balancer module, to check if the read suggestions are valid, are added in osdmap & to confirm the read suggestions are enforced on the clusters.

harshkumarRH commented on ceph/rados/core_workflows.py:
```diff
@@ -541,10 +541,6 @@ def create_pool(self, pool_name: str, **kwargs) -> bool:
             )
             self.node.shell([enable_app_cmd])
 
-        if kwargs.get("app_name") == "rbd":
-            pool_init = f"rbd pool init -p {pool_name}"
-            self.node.shell([pool_init])
```
Any reason for removing this block of code?

harshkumarRH replied:
Ignore, I see it was replaced not removed

harshkumarRH commented on ceph/rados/core_workflows.py:
```diff
@@ -1194,10 +1261,15 @@ def create_erasure_pool(self, name: str, **kwargs) -> bool:
-        profile_name = kwargs.get("profile_name", f"ecp_{name}")
-        rule_name = f"rule_{name}"
+        pool_name = kwargs.get("pool_name", "name")
+        if not pool_name:
+            log.error("No name provided. Exiting")
+            return False
```
Will this block ever get executed?

pdhiran replied:
Hopefully, it would never reach here, Just added this as a failsafe, so that we can fail tests.

pdhiran commented on ceph/rados/core_workflows.py:
```diff
@@ -1112,9 +1115,36 @@ def delete_pool(self, pool: str) -> bool:
+                        return False
+
+                log.info("Deleting the crush rule used for the EC pool")
+                rule_list = self.run_ceph_command(
```
Updated.

harshkumarRH commented on tests/rados/pool_tests.py:
```diff
@@ -1439,6 +1408,132 @@ def run(ceph_cluster, **kw):
+            try:
+                rados_obj.create_erasure_pool(**profile_configs, force=True)
+            except Exception as err:
+                log.warning(f"Modification failed with --force flag. Exception: {err}")
```
If it is expected that the profile gets modified, why are we not stopping the test?

harshkumarRH commented on tests/rados/pool_tests.py:
```diff
@@ -1439,6 +1408,132 @@ def run(ceph_cluster, **kw):
+                    except Exception as err:
+                        log.info(
+                            f"Failed to modify profile with --force and --yes-i-really-mean-it flags. Error: {err}"
+                        )
```
Should we not fail the test at this point?

pdhiran replied:
NO, we will not fail test here, and fail only if the profile was not actually not modified. 

Also, I was expecting the command failure to raise an exception , which is not happening. So we do not enter the except block. I have kept the except block just in case.

pdhiran commented:
> @pdhiran , Please provide the bugzilla and tracker details for the following task.
> 
> Automated 3 bugzilla & tracker scenarios in Online reads balancer module, to check if the read suggestions are valid, are added in osdmap & to confirm the read suggestions are enforced on the clusters.

Added the bugzilla details.

neha-gangadhar reviewed (approved):
Approved
